server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/compaction/files/SelectedUserCompFiles.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.tserver.tablet.compaction.files;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.accumulo.core.metadata.StoredTabletFile;
+import org.apache.accumulo.core.spi.compaction.CompactionKind;
+
+import com.google.common.base.Preconditions;
+
+public class SelectedUserCompFiles implements CompactionFiles {
+  private final Set<StoredTabletFile> allCompactingFiles;
+  private final Set<StoredTabletFile> selectedFiles;
+  private final CompactionKind selectKind;
+
+  public SelectedUserCompFiles(Set<StoredTabletFile> allCompactingFiles,
+      Set<StoredTabletFile> selectedFiles, CompactionKind selectKind) {
+    this.allCompactingFiles = allCompactingFiles;
+    this.selectedFiles = selectedFiles;
+    this.selectKind = selectKind;
+  }
+
+  @Override
+  public Set<StoredTabletFile> getCandidates(Set<StoredTabletFile> currFiles, CompactionKind kind) {
+    if (selectKind == kind) {
+      Set<StoredTabletFile> candidates = new HashSet<>(selectedFiles);
+      candidates.removeAll(allCompactingFiles);
+      candidates = Collections.unmodifiableSet(candidates);
+      Preconditions.checkState(currFiles.containsAll(candidates),
+          "selected files not in all files %s %s", candidates, currFiles);
```

Comment:
This message could be changed similar to the improvements in #2243
```suggestion
          "Selected files: %s not in set of all files: %s", candidates, currFiles);
```